ModelBundle/Repository/SiteRepository.php

```diff
@@ -118,13 +118,17 @@ public function findByAliasDomain($domain)
             )
         ));
 
+        $ids = array();
         if (is_array($commandResult) && array_key_exists('ok', $commandResult) && $commandResult['ok'] == 1) {
             foreach ($commandResult['results'] as $siteId) {
-                return $this->findOneBySiteId($siteId['_id']);
+                $ids[] = $siteId['_id'];
```

Comment:
You were right. KernelExceptionSubscriber waits for a collection, but return, after parse, only one site. As we use mapReduce, the parse in KernelExceptionSubscriber could be done directly in js mongo map and the findByAliasDomain method returns one  SiteInterface or null by becoming a  findByAliasPrefixAndDomain.